modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -439,6 +448,34 @@ INSFVRhieChowInterpolator::finalize()
 #endif
 }
 
+void
+INSFVRhieChowInterpolator::ghostADataOnBoundary(const BoundaryID boundary_id)
+{
+  if (_a_data_provided || this->n_processors() == 1 ||
+      _velocity_interp_method == Moose::FV::InterpMethod::Average)
+    return;
+
+  // Ghost a for the elements on the boundary
+  for (auto elem_id : _moose_mesh.getBoundaryActiveLocalElemIds(boundary_id))
+  {
+    const auto & elem = _moose_mesh.elemPtr(elem_id);
+    if (elem->processor_id() != this->processor_id())
```

Comment:
You should assert that these processor IDs are equal

Reply:
but they are not?

Reply:
yeah this line was not doing anything, might as well delete

Reply:
well actually. getBoundaryActiveLocalElemIds actually gives you local + ghosted
so it can happen

I think we cant have the assert here

Reply:
We need a name for locally owned + ghosted. 
Or should we just say local = local owned + ghosted

Reply:
The API name is totally wrong if it returns local + ghosted. Local is established to mean "owned by this processor" in libMesh and in almost all framework code. I would say all but clearly there is an aberration here

Reply:
it does.
The code works as expected though, the name is the problem imo

Reply:
names changed